articles/models.py

```diff
@@ -262,17 +265,35 @@ class Sharelinks(models.Model):
     cached_facebook_count = models.IntegerField(default=0)
     cached_last_updated = models.DateTimeField(blank=True, null=True)
 
-    def update_cache(self):
-        if not self.cached_last_updated or (timezone.now() - self.cached_last_updated) > timedelta(minutes=10):
+    def _get_twitter_count(self):
+        try:
             url = 'https://cdn.api.twitter.com/1/urls/count.json?url=http://opencanada.org' + self.url
             response = requests.get(url)
```

Comment:
When is this run? If it is during a request response cycle of any kind than make sure gunicorn is configured to use async workers or twitter api not working could bring the open canada site to a crawl.